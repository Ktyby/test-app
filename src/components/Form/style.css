.form {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.form__input {
  width: 200px;
  height: 30px;

  margin-bottom: 10px;
  padding: 0 10px;

  border: none;
  border-radius: 8px;

  background-color: #fcd911;
  color: #000;

  cursor: pointer;
}

.form__button {
  width: 120px;
  height: 30px;

  margin-top: 20px;

  border: none;
  border-radius: 10px;

  background-color: #fcd911;
  cursor: pointer;

  transition: 0.4s;
}

.form__button:hover {
  background-color: #000;
  color: #fff;
}
