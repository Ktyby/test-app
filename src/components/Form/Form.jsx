import React from "react";
import PropTypes from "prop-types";

import "./style.css";

const Form = () => (
  <form className="form">
    <input
      required={true}
      className="form__input"
      type="text"
      name="name"
      placeholder="name"
    />
    <input
      required={true}
      className="form__input"
      type="text"
      placeholder="surname"
      name="surname"
    />
    <input
      required={true}
      className="form__input"
      type="email"
      name="email"
      placeholder="email"
    />
    <input
      required={true}
      className="form__input"
      type="date"
      name="birthday"
    />
    <button className="form__button" type="submit">
      Save
    </button>
  </form>
);

Form.propTypes = {};

Form.defaultProps = {};

export default Form;
