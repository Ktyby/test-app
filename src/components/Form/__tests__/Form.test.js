import renderer from "react-test-renderer";
import React from "react";

import Form from "../Form";

describe("components => Form", () => {
  it("should render Form component", () => {
    const component = renderer.create(<Form />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
