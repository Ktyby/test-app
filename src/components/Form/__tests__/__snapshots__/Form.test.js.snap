// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components => Form should render Form component 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="form__input"
    name="name"
    pattern="^[A-Za-z]+$"
    placeholder="name"
    required={true}
    title="You can use only latin letters for the name"
    type="text"
  />
  <input
    className="form__input"
    name="surname"
    pattern="^[A-Za-z]+$"
    placeholder="surname"
    required={true}
    title="You can use only latin letters for the name"
    type="text"
  />
  <input
    className="form__input"
    name="email"
    placeholder="email"
    required={true}
    type="email"
  />
  <input
    className="form__input"
    name="birthday"
    required={true}
    type="date"
  />
  <button
    className="form__button"
    type="submit"
  >
    Save
  </button>
</form>
`;
